src/try_with_timeout.jl

```diff
@@ -75,15 +75,25 @@ end
 """
 function try_with_timeout(f, timeout, ::Type{T}=Any) where {T}
     ch = Channel{T}(0)
-    x = TimedOut(ch)
-    timer = Timer(tm -> !isready(ch) && close(ch, TimeoutException(timeout)), timeout)
+    timer = Timer(timeout)
     @samethreadpool_spawn begin
+        _ch = $ch
         try
-            put!(ch, $f(x)::T)
+            wait($timer)
+            !isready(_ch) && close(_ch, TimeoutException($timeout))
         catch e
-            close(ch, CapturedException(e, catch_backtrace()))
+            e isa EOFError || close(_ch, CapturedException(e, catch_backtrace()))
+        end
+    end
```

Comment:
Ugh, I agree this is right, but we really need an encapsulation of this pattern. This turned 1 line into 9 lines that basically do the same thing, to work around a bug in julia's scheduler.

Either we need to fix the issues with julia to make this not pin the parent task (but i know that's hard / maybe impossible until a 2.0 release), or maybe this package should also provide a timer-do abstraction to encapsulate this pattern?